support/modules/single-file

```diff
@@ -29,22 +40,22 @@ case "$STATE" in
         mkdir -p "${tmp_dest_dir}"
         if test -f "${dest_dir}/${filename}"
         then
-            if ! cp -a "${dest_dir}/${filename}" "${tmp_dest_dir}"
+            if ! safe_copy "${dest_dir}/${filename}" "${tmp_dest_dir}"
             then
                 ret=$?
                 # Make sure there is no half-backup lying around.
                 rm -rf "${tmp_dest_dir}"
                 exit $ret
             fi
         fi
-        cp "$FILES"/files/"${filename}" "${dest_dir}/${filename}"
         # Previous revisions of this update module did not use the
         # permissions_file, so it might not exist.
         if test -f "${permissions_file}"
         then
             mode=$(cat "${permissions_file}")
-            chmod "${mode}" "${dest_dir}/${filename}"
+            chmod "${mode}" "$FILES"/files/"${filename}"
```

Comment:
Question:
Was this wrong previously?

Reply:
Same reason: It is not atomic. There is a window in time where the file has been replaced, but it is not executable. It would be possible to do the `chmod` inside `safe_copy` instead of in advance here, but this method avoids having a special case for executables inside `safe_copy`, and does the same job.

Reply:
I think I'd prefer the special case handling, but all yours

Reply:
I guess we agree to disagree then! ;-)
It's partially because `safe_copy` actually comes from mender-configure, and it's nice if it's reusable.